ANGLE is compiled with EAGL on iOS family

kkinnunen-apple opened the pull request:
#### 06284a95d3df3ceca9cdd1190f466e157441a686
<pre>
ANGLE is compiled with EAGL on iOS family
<a href="https://bugs.webkit.org/show_bug.cgi?id=263637">https://bugs.webkit.org/show_bug.cgi?id=263637</a>
rdar://117460577

Reviewed by Alexey Proskuryakov.

Avoid compiling ANGLE with EAGL support on iOS family.
All iOS family devices ship with Metal.

* Source/ThirdParty/ANGLE/Configurations/ANGLE-dynamic.xcconfig:

Canonical link: <a href="https://commits.webkit.org/269806@main">https://commits.webkit.org/269806@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/afc8bbfbba96f5ed415732c994198f21e2ef01d6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/23632 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/1748 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/24753 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/25791 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/21815 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/3334 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/24168 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/22380 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/23875 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/1306 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/20447 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/26387 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/1094 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/21353 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/27628 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/21560 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/21622 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/25389 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/1060 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/18764 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/1039 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5647 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/1473 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/1350 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/ThirdParty/ANGLE/Configurations/ANGLE-dynamic.xcconfig:
```diff
@@ -33,8 +33,7 @@ GCC_PREPROCESSOR_DEFINITIONS = $(DEBUG_DEFINES) LIBGLESV2_IMPLEMENTATION LIBANGL
 GCC_PREPROCESSOR_DEFINITIONS_maccatalyst[arch=arm*] = $(GCC_PREPROCESSOR_DEFINITIONS_macosx) ANGLE_ENABLE_EAGL GLES_SILENCE_DEPRECATION ANGLE_CPU_ARM64
-// FIXME: Rest of the iOS family does not need EAGL, these will be removed when upstream is able to compile without it.
-GCC_PREPROCESSOR_DEFINITIONS_iphoneos = ANGLE_ENABLE_GLSL ANGLE_ENABLE_ESSL ANGLE_ENABLE_OPENGL ANGLE_ENABLE_APPLE_WORKAROUNDS ANGLE_ENABLE_EAGL GLES_SILENCE_DEPRECATION
+GCC_PREPROCESSOR_DEFINITIONS_iphoneos =
```
We don't need an empty definition (nor the lines below, these should all be removed now). They do nothing other than inviting questions like "where's visionOS?"

Secondly, the name is not great - `GCC_PREPROCESSOR_DEFINITIONS_maccatalyst` implies that this is the whole value of `GCC_PREPROCESSOR_DEFINITIONS` for the platform, not just platform-specific ANGLE additions.

kkinnunen-apple committed:
ANGLE is compiled with EAGL on iOS family
https://bugs.webkit.org/show_bug.cgi?id=263637
rdar://117460577

Reviewed by Alexey Proskuryakov.

Avoid compiling ANGLE with EAGL support on iOS family.
All iOS family devices ship with Metal.

* Source/ThirdParty/ANGLE/Configurations/ANGLE-dynamic.xcconfig:

Canonical link: https://commits.webkit.org/269806@main
